util/internal/test/src/main/java/org/hibernate/search/util/impl/test/rule/log4j/LogChecker.java

```diff
@@ -43,7 +43,8 @@ public void appendFailure(Description description, String newline) {
 		}
 	}
 
-	void process(LogEvent event) {
+	// This must be synchronized to avoid problems when multiple threads issue log events concurrently
+	synchronized void process(LogEvent event) {
```

Comment:
It seems that this method only appends data to `extraEvents`, there is no remove.
Moreover, possible interleaving of `add` on `extraEvents` and count increment would not affect the consistency, or, at least, I think so.
I was asking myself if we could remove the `synchronized` here, with some changes:
 
1. Create the event list in the constructor
2. Use an atomic integer for the counter
3. Use a concurrent friendly data structure for the event lists, such as `ConcurrentLinkedQueue`. 
 
I know that this is test code (not production code), but maybe it is worth trying to make them more efficient.

Reply:
I don't think this is worth optimizing, especially considering the additional complexity the optimization would introduce.